swarm/testutil/http.go

```diff
@@ -49,24 +62,64 @@ func NewTestSwarmServer(t *testing.T) *TestSwarmServer {
 		ChunkStore: localStore,
 	}
 	dpa.Start()
-	a := api.NewApi(dpa, nil)
+
+	// mutable resources test setup
+	resourceDir, err := ioutil.TempDir("", "swarm-resource-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rh, err := storage.NewResourceHandler(resourceDir, &testCloudStore{}, &fakeBackend{}, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	a := api.NewApi(dpa, nil, rh)
 	srv := httptest.NewServer(httpapi.NewServer(a))
 	return &TestSwarmServer{
 		Server: srv,
 		Dpa:    dpa,
 		dir:    dir,
+		hasher: storage.MakeHashFunc(storage.SHA3Hash)(),
+		cleanup: func() {
+			srv.Close()
+			rh.Close()
+			dpa.Stop()
+			os.RemoveAll(dir)
+			os.RemoveAll(resourceDir)
+		},
 	}
 }
 
 type TestSwarmServer struct {
 	*httptest.Server
-
-	Dpa *storage.DPA
-	dir string
+	hasher  storage.SwarmHash
+	Dpa     *storage.DPA
+	dir     string
+	cleanup func()
 }
 
 func (t *TestSwarmServer) Close() {
-	t.Server.Close()
-	t.Dpa.Stop()
-	os.RemoveAll(t.dir)
+	t.cleanup()
+}
+
+type testCloudStore struct {
+}
+
+func (c *testCloudStore) Store(*storage.Chunk) {
+}
+
+func (c *testCloudStore) Deliver(*storage.Chunk) {
+}
+
+func (c *testCloudStore) Retrieve(*storage.Chunk) {
+}
+
+// for faking the rpc service, since we don't need the whole node stack
+type FakeRPC struct {
+	blocknumber uint64
+}
+
+func (r *FakeRPC) BlockNumber() (string, error) {
+	return strconv.FormatUint(r.blocknumber, 10), nil
```

Comment:
Is this being used?

Reply:
yes, it's provides the fake block number retriever, same as in `swarm/storage/resource_test.go`

Reply:
It doesn't seem to be instantiated anywhere.

Reply:
It is in line 76, when creating the `ResourceHandler`

Reply:
That's `fakeBackend` not `FakeRPC`

Reply:
right you are, thanks